components/clarinet-deployments/src/lib.rs

```diff
@@ -454,14 +449,23 @@ pub async fn generate_default_deployment(
                 Ok(inferable_dependencies) => {
                     // Looping could be confusing - in this case, we submitted a HashMap with one contract, so we have at most one
                     // result in the `inferable_dependencies` map. We will just extract and keep the associated data (source, ast, deps).
-                    for (contract_id, dependencies) in inferable_dependencies.into_iter() {
+                    if let Some((contract_id, dependencies)) =
+                        inferable_dependencies.into_iter().next()
+                    {
                         for dependency in dependencies.iter() {
                             queue.push_back((dependency.contract_id.clone(), None));
                         }
                         requirements_deps.insert(contract_id.clone(), dependencies);
                         requirements_data.insert(contract_id.clone(), (clarity_version, ast));
-                        break;
                     }
+                    // for (contract_id, dependencies) in inferable_dependencies.into_iter() {
+                    //     for dependency in dependencies.iter() {
+                    //         queue.push_back((dependency.contract_id.clone(), None));
+                    //     }
+                    //     requirements_deps.insert(contract_id.clone(), dependencies);
+                    //     requirements_data.insert(contract_id.clone(), (clarity_version, ast));
+                    //     break;
+                    // }
```

Comment:
@lgalabru 
I left that commented for now, do you think the new version works the same way?
It looks like it would always break after the first iteration right?
Or did I miss something?